<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<title>[Jack-Devel] Calling jack_cycle_wait/jack_cycle_signal from a "third party" thread.</title>
<style>
pre {background: #fafafa; border: 1px solid #f1f2f3; font-size: 1.2em; padding: 10px; white-space: pre-wrap; white-space: -moz-pre-wrap; white-space: -pre-wrap; white-space: -o-pre-wrap; word-wrap: break-word;}
h1 {font-size: 140%;}
.header_table {table-layout: fixed; width: 100%;}
.col1 {width: 120px; vertical-align: top;}
.from {font-weight: bold;}
</style>
</head>
<body>
<h1>[Jack-Devel] Calling jack_cycle_wait/jack_cycle_signal from a "third party" thread.</h1>
<table class="header_table">
<tr><td class='col1'><a href='../1437470748.28623_0.ltw:2,/index.html'>Prev</a></td><td><a href='../1439078187.13844_0.ltw:2,/index.html'>Next</a>&nbsp;&nbsp;<a href='../../index.html#1439045695.9942_0.ltw:2,'>Index</a></td></tr>
<tr><td class='col1'>Date</td><td>Sat, 08 Aug 2015 16:54:47 +0200 </td></tr>
<tr><td class='col1'>From</td><td><span class="from"> Kjetil Matheussen </span> &lt;[hidden] at gmail dot com&gt; </td></tr>
<tr><td class='col1'>To</td><td>"[hidden] at lists dot jackaudio dot org" &lt;[hidden] at lists dot jackaudio dot org&gt; </td></tr>
<tr><td class='col1'>Follow-Up</td><td>Paul Davis
<a href='../1439078187.13844_0.ltw:2,/index.html'>Re: [Jack-Devel] Calling jack_cycle_wait/jack_cycle_signal from a "third party" thread. </a></td></tr>
<tr><td class='col1'>Follow-Up</td><td>Robin Gareus
<a href='../1439162653.7812_0.ltw:2,/index.html'>Re: [Jack-Devel] Calling jack_cycle_wait/jack_cycle_signal from a "third party" thread. </a></td></tr>
</table>
<pre>
I.e. the question is about calling jack_cycle_wait/jack_cycle_signal from
a thread that is NOT created with jack_set_process_thread.

Should that be okay? I don't see anything about it in the documentation.

I'm doing this as a workaround until I figure out why on earth
the global exception handler on Windows isn't called when
crashing in the jack thread. (A "global exception handler" in
windows is called "unhandled exceptions filter", and it is created
with the "SetUnhandledExceptionFilter" call)

As far as I can see, jack on windows does not implement any type of
Windows exceptions handler (it's a quite complicated system), except
for doing some C++ try/catch stuff, but at least if you
do try/catch stuff in mingw, it doesn't screw up the unhandled
exceptions filter. Besides, I didn't find any try/catch code for the
process thread either. So I have no idea what happens...

I've also tried crashing the jack process thread immediately after
it starts, to eliminate the possibility that it's my code that does
something strange, but the unhandled exceptions filter isn't called then
either.
</pre>
<table class="header_table">
<tr><td class='col1'><a href='../1437470748.28623_0.ltw:2,/index.html'>Prev</a></td><td><a href='../1439078187.13844_0.ltw:2,/index.html'>Next</a>&nbsp;&nbsp;<a href='../../index.html#1439045695.9942_0.ltw:2,'>Index</a></td></tr>
</table>
<p><small>1439045695.9942_0.ltw:2,&nbsp;&lt;CAC6niE+p9bxRS2uJGM=0eaA6RZLaDfdCx+y=WnxMWiDaqZiMAg at mail dot gmail dot com&gt;</small></p>
<!-- Created with Bash Archive Mail -->
</body>
</html>
